src/Fixer/StringNotation/ExplicitStringVariableFixer.php

```diff
@@ -59,17 +67,54 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
             }
 
             $prevToken = $tokens[$index - 1];
-            if (
-                $prevToken->isGivenKind(T_ENCAPSED_AND_WHITESPACE)
-                || $prevToken->isGivenKind(T_START_HEREDOC)
-                || '"' === $prevToken->getContent()
-            ) {
+            if (!$this->isStringPartToken($prevToken)) {
+                continue;
+            }
+
+            $variableTokens = [
+                $index => $token,
+            ];
+            $firstVariableTokenIndex = $index;
+            $lastVariableTokenIndex = $index;
+
+            $nextIndex = $index + 1;
+            while (!$this->isStringPartToken($tokens[$nextIndex])) {
+                $variableTokens[$nextIndex] = $tokens[$nextIndex];
+                $lastVariableTokenIndex = $nextIndex;
+                ++$nextIndex;
+            }
+
+            if (1 === count($variableTokens)) {
                 $tokens->overrideRange($index, $index, [
                     new Token([T_DOLLAR_OPEN_CURLY_BRACES, '${']),
                     new Token([T_STRING_VARNAME, substr($token->getContent(), 1)]),
                     new Token([CT::T_DOLLAR_CLOSE_CURLY_BRACES, '}']),
                 ]);
+            } else {
+                foreach ($variableTokens as $variablePartIndex => $variablePartToken) {
+                    if ($variablePartToken->isGivenKind(T_NUM_STRING)) {
+                        $tokens[$variablePartIndex] = new Token([T_LNUMBER, $variablePartToken->getContent()]);
+                    }
+                }
+
+                $tokens->insertAt($lastVariableTokenIndex + 1, new Token([CT::T_CURLY_CLOSE, '}']));
+                $tokens->insertAt($firstVariableTokenIndex, new Token([T_CURLY_OPEN, '{']));
```

Comment:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3327#discussion_r157570814

Reply:
I think you've reached the hundredth time you are talking about going backward. Maybe we should write an AutoReview?

BTW, fixed.

Reply:
it's not always needed to go backward. also, even if adding new tokens to collection, going backward is not only solution.